### PR TITLE
hw-6-2

### PR DESCRIPTION
**Pull request by kenu21**

### CODE REVIEW

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 1]
```diff
@@ -2,15 +2,45 @@
 
 import core.basesyntax.Storage;
 
+import java.util.Arrays;
+
 public class StorageImpl<K, V> implements Storage<K,V> {
 
+    private Object[] keys = new Object[16];
+    private Object[] values =  new Object[16];
+    int topArray = -1;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/impl/StorageImpl.java]: add access modifier

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 2]
```diff
@@ -2,15 +2,45 @@
...
+
     @Override
     public void put(K key, V value) {
+        K k;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/impl/StorageImpl.java]: do we need this variable?

File: src/main/java/core/basesyntax/impl/StorageImpl.java (hunk 1 above)
Comment (Sofasmile) [inline src/main/java/core/basesyntax/impl/StorageImpl.java]: why -1?

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 3]
```diff
@@ -2,15 +2,45 @@
...
+                values[i] = value;
+                return;
+            }
+        }
+        topArray++;
+        if (topArray < keys.length - 1) {
+            keys[topArray] = key;
+            values[topArray] = value;
+        } else {
+            keys = Arrays.copyOf(keys, keys.length * 2);
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/impl/StorageImpl.java]: better move this logic in the separate private method

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 4]
```diff
@@ -2,15 +2,45 @@
...
+            values = Arrays.copyOf(keys, keys.length * 2);
+            keys[topArray] = key;
+            values[topArray] = value;
+        }
     }
 
     @Override
     public V get(K key) {
+        K k;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/impl/StorageImpl.java]: the same

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 5]
```diff
@@ -2,15 +2,45 @@
...
 public class StorageImpl<K, V> implements Storage<K,V> {
 
+    private Object[] keys = new Object[16];
+    private Object[] values = new Object[16];
+    private int topArray = 0;
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < keys.length; i++) {
+            if ((K) keys[i] == key) {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/impl/StorageImpl.java]: how will work this method if key is object?
what is the difference between equals and ==?

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 6]
```diff
@@ -2,15 +2,45 @@
...
+                values[i] = value;
+                return;
+            }
+        }
+        if (topArray < keys.length - 1) {
+            keys[topArray] = key;
+            values[topArray] = value;
+        } else {
+            keys = resizeAndCopyArr(keys);
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/impl/StorageImpl.java]: better remove line 25 and 26 in method resizeAndCopyArr()
and then your code will look:

resizeAndCopyArr();
keys[topArray] = key;
values[topArray] = value;
topArray++;

File: src/main/java/core/basesyntax/impl/StorageImpl.java (hunk 5 above)
Comment (kenu21) [reply src/main/java/core/basesyntax/impl/StorageImpl.java]: We will compare value of link.
We can compare value of link for objects with == and objects with equals. We can compare values primitives with ==.

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 7]
```diff
@@ -2,15 +2,45 @@
...
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < keys.length; i++) {
+            if (keys[i] != null && keys[i].equals(key)) {
+                values[i] = value;
+                return;
+            }
+        }
+        if (topArray < keys.length - 1) {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/impl/StorageImpl.java]: we can simplify:
if (topArray >= keys.length - 1) {
resizeAndCopyArr(keys);
            resizeAndCopyArr(values);
}
            keys[topArray] = key;
            values[topArray] = value;
        topArray++;

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 8]
```diff
@@ -2,15 +2,45 @@
...
 public class StorageImpl<K, V> implements Storage<K,V> {
 
+    private Object[] keys = new Object[16];
+    private Object[] values = new Object[16];
+    private int topArray = 0;
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < keys.length; i++) {
+            if (keys[i] != null && keys[i].equals(key)) {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/impl/StorageImpl.java]:
```suggestion
            if (keys[i] == key || keys[i].equals(key)) {
```

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 9]
```diff
@@ -2,15 +2,45 @@
...
+            keys[topArray] = key;
+            values[topArray] = value;
+        }
+        topArray++;
     }
 
     @Override
     public V get(K key) {
+        for (int i = 0; i < keys.length; i++) {
+            if (keys[i] == key) {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/impl/StorageImpl.java]:
```suggestion
            if (keys[i] == key || keys[i].equals(key)) {
```

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 10]
```diff
@@ -6,41 +6,39 @@
 
 public class StorageImpl<K, V> implements Storage<K,V> {
 
-    private Object[] keys = new Object[16];
-    private Object[] values = new Object[16];
+    private static final int ARRAYSIZE = 10;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/impl/StorageImpl.java]: ARRAY_SIZE

**Pull request closed without merging**